tooling/minimize/tests/ub/tree_borrows/array_access.rs

```diff
@@ -0,0 +1,20 @@
+extern crate intrinsics;
+use intrinsics::*;
+
+//@ compile-flags: --minimize-tree-borrows
+fn main() {
+    unsafe {
+        let mut data = [0u8, 1, 2, 3];
+        let x = &mut data[0] as *mut u8; // (x, [R, R, R, R])
+        let y = &mut *x; // (x, [R, R, R, R]) -> (y, [R, R, R, R])
+        let yraw = y as *mut u8;
+
+        *x.add(1) = 42; // (x, [R, A, R, R]) -> (y, [R, D, R, R])
+
+        print(*yraw.add(0));
```

Comment:
```suggestion
        // y can still be used for all the other offsets
        print(*yraw.add(0));
```
In fact it could even be used to write, right? Maybe make these writes instead of reads.